src/chunk_append/planner.c

```diff
@@ -102,7 +102,21 @@ chunk_append_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *path, L
 	cscan->custom_scan_tlist = tlist;
 	cscan->scan.plan.targetlist = tlist;
 
-	if (path->path.pathkeys != NIL)
+	if (path->path.pathkeys == NIL)
+	{
+		ListCell *lc_plan, *lc_path;
```

Comment:
we push down target list in either case but when there are pathkeys we have to make sure order columns are in targetlist in case we need to inject sort node